build.tcl

```diff
@@ -335,15 +344,19 @@ proc _install {args} {
     }
 
     if {[llength $args] == 0} {
-	set exe  [file dirname [file normalize [info nameofexecutable]/___]]]
+	set exe  [file dirname [file normalize [info nameofexecutable]/___]]
 	set dstl [info library]
 	set dsta [file dirname $exe]
-	set dsti [file dirname $dsta]/include
+	set dsti [file join [file dirname $dsta] include]
 
     } else {
 	set dstl [lindex $args 0]
-	set dsta [file dirname [findlib $dstl]]/bin
-	set dsti [file dirname [findlib $dstl]]/include
+	if {![file exists [file dirname $dstl]]} {
+	    file mkdir $dstl/../bin
+	    file mkdir $dstl/../include
```

Comment:
Should we use `[file dirname $dstl]` ? (Maybe even compute once, before the `if` check, and reuse in all three places (if check, and the mkdir commands)).

Reply:
And of course `file join`, for consistency.

Reply:
I've already changed a bit of this code so that it makes more sense.

My opinion is that "/" should be preferred over [file join], which is only needed in cases where a component after the first component could be an absolute path that should override the first component.  There are no such cases in build.tcl, so I'd like to convert remaining occurrences of [file join] to "/".

The build went wrong in such a way that one file was overwriting another, so a test that expected files are produced when build to an alternate prefix would be sufficient.


Perhaps the first argument to [build.tcl install] should be the path to a prefix to install in, rather than the libdir, in which case these bits of code would be rewritten.

Reply:
[file join] also helps eliminate redundant file separators in case a prefix is "/".  I'd like to  use [file join] in these cases as well, but use "/" where [file join] adds no value.

Reply:
After reviewing the documentation for on file names I'm of the opinion that any use of "/" in a Tcl script to separate file name components is ill-advised.  I'm planning now to remove all such occurrences from build.tcl.

Reply:
Ok. Ping me when it is done and I will have another look.